### ISSUE TITLE
Make an ability to set max size of files for downloading

### ISSUE THREAD

**Issue by MasterPetrik** [Bionus/imgbrd-grabber#965]
got from this https://github.com/Bionus/imgbrd-grabber/issues/954 issue
>  This "limit" depends on the computer, OS, etc. I wouldn't want to have users who before could download files have them un-downloadable after the update.

Then obviously you need to make an option in settings to set this max size, set it by default by for example 50-100MB, so if users have no problem with big files, they could increase this number if they needed, and users who DON'T want to download big files at all could set it for example at 2 MB if they need only low weight images etc. Anyway such option very needed even not because of  https://github.com/Bionus/imgbrd-grabber/issues/954 bug, but It's a good option by itself

**Comment by Bionus** [comment]
The point is that I only added this limit to prevent inevitable crashes, but in the long run (i.e. once suggestion #963 is implemented), this limit will disappear, to only be used for image display in the image window.

So setting a let's say 50MB size limit by default would, once said suggestion is implemented, block the download of normally downloadable images.

> users who DON'T want to download big files at all could set it for example at 2 MB if they need only low weight images etc.

So that would be separate from the crash issue then (preventing crashes v. not wanting to download big images). Kind of applying the "filesize:<2MB" post-filtering to all searches/downloads with a filesize set in the options?

Isn't that too specific? Wouldn't allowing to apply a default post-filtering + allowing post-filtering during batch downloads be an even more powerful feature? Then people wishing to ignore all images below 2MB could use it, but it could also work for people wishing to ignore all images smaller than a given size, or whatever post-filtering allows (source, rating, id, width, height, filesize, score, mpixels, date).

**Comment by Bionus** [comment]
> P.S. It's you told me to post sepatate issues, so I guess that's ok.

Yes that's much more convenient for tracking things, much better for me :+1:
Else it gives issues that are virtually inclosable such as #473 or #492, as they contain many (interesting) things that are not all (yet) done.

**Comment by MasterPetrik** [comment]
> Else it gives issues that are virtually inclosable such as #473 or #492, as they contain many (interesting) things that are not all (yet) done.

No. No. No.  Not my case. In your example it a all-in-one-bunch issues, where posts are not really connected with each other. In my case there are separated issues for each feature, and if something wrong with that: wrong ot particular realisation \ little or hot bugs \ little hot improvements. Then I rather posted it there, because it strongly connected to this, and not much time

**Comment by MasterPetrik** [comment]
> So that would be separate from the crash issue then

This is already separated feature request) preventing crashes is just an example of usefulness of that feature

> Kind of applying the "filesize:<2MB" post-filtering to all searches/downloads with a filesize set in the options?
> Isn't that too specific? Wouldn't allowing to apply a default post-filtering + allowing post-filtering during batch downloads be an even more powerful feature? 

Fully agreed with that part.

> it could also work for people wishing to ignore all images smaller than a given size, or whatever post-filtering allows (source, rating, id, width, height, filesize, score, mpixels, date).

What filtering would be really needed:
Filesize filtering
Size filtering (height, weight, mpixels)
Date filtering someone requested already 
source filtering - already exist as tag, separate filter needed
rating filtering - already exist as tag, separate filter needed

score filtering - bad idea, but already exist too
id filtering - ...what's that?

# MAIN SUGGESTION OF WHOLE TOPIC:
Would be better add filtering features not in options, but in "+" button menu right to the search field. Current one is too poor to be helpful, so I think you could fully rework it and turn this menu into useful additional options menu for search requests with powerful filtering settings(like it made in some search engines like google)

![2017-06-25_064329 win7 mastercore](https://user-images.githubusercontent.com/22294259/27513594-83357c1e-5974-11e7-95b0-dcb27e81bbe3.png)

**Comment by Bionus** [comment]
But isn't that exactly what "post-filtering" is? 😕
Putting "-filesize:<2MB" in that field will remove all files bigger than 2MB.

Or are you suggesting to split it into lots of small fields? That would make the "+" menu very big very fast: checkbox to invert filter ("-" prefix), select box for the type of filter (or one line per possible filter), select box for the operator (">", ">="), value. That would be even more complicated with ranges (ex: "id:10..20").

Maybe adding a "Plus" window to help generating this field (as the syntax is not always very obvious) could help? Kind of like the "Plus" button near the search field?

**Comment by MasterPetrik** [comment]
I mean search filtering like it made for example in Google
 ![2017-06-26_041544 win7 mastercore](https://user-images.githubusercontent.com/22294259/27545989-077833ce-5a9a-11e7-8e0c-efe28d056037.png)

Simple for new users and not so complicated

**Comment by ItsSohma** [comment]
Good idea. Seems difficult to implement, but this would make a nice feature.

**Comment by Bionus** [comment]
Something like that?

![image](https://user-images.githubusercontent.com/882719/27763523-516e2008-5e85-11e7-8a01-f231e9d0d355.png)

**Comment by MasterPetrik** [comment]
Indeed, but some minor important notes:
1. All possible fields should be filled by default with Max and Min default values in Grabber to show what it should look like.
2. Date field should have an option to set date via calendar, I recommend you also make this option like it made on Google.
3. Rating field shouldn't be empty, for default it should be in "All" position
4. We don't know how do you Interpret filesize and other values(Is this value in Kb? Mb? Gb?) so for size and filesize fields you need to write what units do they use.
5. It's not really correct to write "min" and "max" with date field,it need to be like "From" and "To". And "Date" is a bad description, it's not very clear, I guess "Posting date" or "Date range" would be better.
6. Isn't the main search field already consist tags field?
7. Need a default values\reset button

**Comment by Bionus** [comment]
1. Grabber don't use any "default" values for filters, they're just disabled. For most, the equivalent would be min=0, max=+inf. I'm definitely not sure that'd be clearer.
2. :+1:
3. :+1:
4. Depends on what you write, actually. if you don't put any unit, it's bytes, else it will use whatever you type (ex: min=5KB, max=10MB).
5. I'll go with "Posting date" or equivalent then.
6. Unless the source limits the maximum number of tags you can enter.
7. :+1:

**Comment by MasterPetrik** [comment]
> Grabber don't use any "default" values for filters, they're just disabled. For most, the equivalent would be min=0, max=+inf. I'm definitely not sure that'd be clearer.

Then you need to write some max limit anyway, infinity is always less clearer then any value, and the empty field is even more unclear for users

> Depends on what you write, actually. if you don't put any unit, it's bytes, else it will use whatever you type (ex: min=5KB, max=10MB).

Ho-ho! I would like to see to user, who will write images sizes in bytes... Or you write a proper describtion of using this field, or set min and max values, because this is VERY UNCLEAR, I recommend you set min at 1KB and max at 1000MB or something, because there is no users who will write units in this field manually, if they don't know about it.

> I'll go with "Posting date" or equivalent then.

And think about "past 24 hours" "past week" and "past mounth" options for date field, it's very useful for many users, and custom range option, only with it show min and max date fields(like in google, I think it's the best way)

**Comment by ItsSohma** [comment]
If you're describing fields rather than entering defaults, you could use a little i icon for info about it or a question mark for help and then write a description like for max: Enter a number greater than or equal to 1 to limit the return.
